File: .tmp_py4013/batch_12_dump/3407_substring_matching_pattern.js
// LeetCode 3407 - Substring Matching Pattern
// https://leetcode.com/problems/substring-matching-pattern/

var hasMatch = function(s, p) {
    const i = p.indexOf('*');
    const left = p.substring(0, i);
    const right = p.substring(i + 1);
    const li = s.indexOf(left);
    if (li < 0) return false;
    return s.indexOf(right, li + left.length) >= 0;
};

File: .tmp_rs4013/rust_check17/3407_substring_matching_pattern.rs
struct Solution;
// LeetCode 3407 - Substring Matching Pattern
// https://leetcode.com/problems/substring-matching-pattern/

impl Solution {
    pub fn has_match(s: String, p: String) -> bool {
        let i = p.find('*').unwrap();
        let left = &p[..i];
        let right = &p[i + 1..];
        if let Some(li) = s.find(left) {
            s[li + left.len()..].contains(right)
        } else {
            false
        }
    }
}

fn main() {}
